geometry/shape_specification.cc

```diff
@@ -433,6 +433,105 @@ double CalcVolume(const Shape& shape) {
       }});
 }
 
+template <typename T>
+std::optional<Eigen::Vector3<T>> GetNormalAtPointForBox(
```

Comment:
Similar free functions would be implemented for the rest of Shapes (ie `GetNormalAtPointForSphere`, `GetNormalAtPointForEllipsoid`, etc). First I want to get everything else right at least with Box.